jdaviz/configs/imviz/plugins/links_control/links_control.vue

```diff
@@ -4,45 +4,57 @@
     :link="'https://jdaviz.readthedocs.io/en/'+vdocs+'/'+config+'/plugins.html#link-control'"
     :popout_button="popout_button">
 
-    <v-row>
-      <v-radio-group 
-        label="Link type"
-        hint="Type of linking to be done."
-        v-model="link_type"
-        persistent-hint
-        row>
-        <v-radio
-          v-for="item in link_types"
-          :key="item"
-          :label="item"
-          :value="item"
-        ></v-radio>
-       </v-radio-group>
-    </v-row>
+    <div style="display: grid"> <!-- overlay container -->
+      <div style="grid-area: 1/1">
+        <v-row>
+          <v-radio-group 
+            label="Link type"
+            hint="Type of linking to be done."
+            v-model="link_type"
+            persistent-hint
+            row>
+            <v-radio
+              v-for="item in link_types"
+              :key="item"
+              :label="item"
+              :value="item"
+            ></v-radio>
+           </v-radio-group>
+        </v-row>
 
-    <v-row v-if="false">
-      <v-switch
-        label="Fallback on Pixels"
-        hint="If WCS linking fails, fallback to linking by pixels."
-        v-model="wcs_use_fallback"
-        persistent-hint>
-      </v-switch>
-    </v-row>
+        <v-row v-if="false">
+          <v-switch
+            label="Fallback on Pixels"
+            hint="If WCS linking fails, fallback to linking by pixels."
+            v-model="wcs_use_fallback"
+            persistent-hint>
+          </v-switch>
+        </v-row>
 
-    <v-row v-if="link_type == 'WCS'">
-      <v-switch
-        label="Fast approximation"
-        hint="Use fast approximation for image alignment if possible (accurate to <1 pixel)."
-        v-model="wcs_use_affine"
-        persistent-hint>
-      </v-switch>
-    </v-row>
-
-    <v-row justify="end">
-      <j-tooltip tipid='plugin-link-apply'>
-        <v-btn color="accent" text @click="do_link">Link</v-btn>
-      </j-tooltip>
-    </v-row>
+        <v-row v-if="link_type == 'WCS'">
+          <v-switch
+            label="Fast approximation"
+            hint="Use fast approximation for image alignment if possible (accurate to <1 pixel)."
+            v-model="wcs_use_affine"
+            persistent-hint>
+          </v-switch>
+        </v-row>
+      </div>
+      <div v-if="linking_in_progress"
+           class="text-center"
+           style="grid-area: 1/1; 
+                  z-index:2;
+                  margin-left: -24px;
+                  margin-right: -24px;
+                  padding-top: 60px;
+                  background-color: rgb(0 0 0 / 20%)">
+        <v-progress-circular
+          indeterminate
+          color="spinner"

```

Comment:
If the spinner does not appear for half the users, I wonder if we should just have a text-based status bar. And then we can also post the "linking successful" message there instead of the snackbar message that Harry couldn't find. (And maybe even add a time stamp.)

Reply:
I suspect the problem is more with vue processing the change in traitlet than the spinner itself (since the spinner works from the API), but this is at least a good idea for debugging if that suspicion is true or not!

Reply:
@pllim @duytnguyendtn - can you please try out my `DBG-live-linking-spinner` branch?  In addition to the spinner, it should show a "LINKING IN PROGRESS" message at the top of the plugin - I'm very curious to know whether this shows but the spinner doesn't when you're interacting with the UI.

<img width="993" alt="image" src="https://user-images.githubusercontent.com/877591/187481810-44155e5e-1248-4354-aa40-f8a100b0dc73.png">

Reply:
I tried this branch in a new Python 3.10 env, and the spinner now shows via both plugin GUI and API. 🤷

Reply:
And then I switched back to the older Python 3.9 env and using `DBG-live-linking-spinner` branch, and also see spinner now via both... 🤯

Reply:
Aha! Turns out when spinner works, I had ipywidgets 7.7 installed (for testing #1611). When I upgraded ipywidgets back to 8.0.1, the GUI does not show spinner *nor* "linking in progress" words (none of those show up).